src/main/java/duke/logic/parser/Parser.java

```diff
@@ -0,0 +1,202 @@
+package duke.logic.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.logic.commands.Command;
+import duke.logic.commands.DeadlineCommand;
+import duke.logic.commands.DeleteCommand;
+import duke.logic.commands.DoneCommand;
+import duke.logic.commands.EventCommand;
+import duke.logic.commands.ExitCommand;
+import duke.logic.commands.FindCommand;
+import duke.logic.commands.InvalidCommand;
+import duke.logic.commands.ListCommand;
+import duke.logic.commands.ToDoCommand;
+
+/**
+ * Parses user input.
+ */
+public class Parser {
+
+    /**
+     * Parses user input into command for execution.
+     *
+     * @param text The user input string.
+     * @return The command based on the user input.
+     */
+    public Command parse(String text) {
+        String[] tokens = text.trim().split("\\s+", 2); // split at the first white space
+        Command command = parseTokens(tokens);
+        return command;
+    }
+
+    /**
+     * Categorizes the input into respective command types,
+     * to prepare for further processing.
+     *
+     * @param tokens The tokenized input array.
+     * @return The command based on the user input.
+     */
+    private Command parseTokens(String[] tokens) {
+        switch (tokens[0]) {
+        case "bye":
+            return handleBye(tokens);
+        case "list":
+            return handleList(tokens);
+        case "done":
+            return handleDone(tokens);
+        case "delete":
+            return handleDelete(tokens);
+        case "todo":
+            return handleToDo(tokens);
+        case "deadline":
+            return handleDeadline(tokens);
+        case "event":
+            return handleEvent(tokens);
+        case "find":
+            return handleFind(tokens);
```

Comment:
Should a `//Fallthrough` comment be added in these cases as there is no break statement?

Reply:
I'm not sure if it's needed here, because the function immediately returns in each case so there is no possibility of falling through to a different case.